nntrainer/graph/network_graph.cpp

```diff
@@ -363,13 +368,14 @@ int NetworkGraph::checkCompiledGraph() {
   auto const &l = getSortedLayerNode(0);
   /** First layer cannot be activation, batch normalization or loss */
   const std::string &type = l->getType();
-  if (istrequal(type, ActivationLayer::type) ||
-      istrequal(type, BatchNormalizationLayer::type) ||
-      istrequal(type, LossLayer::type)) {
-    ml_loge("%s cannot be the first layer, type: %s", l->getName().c_str(),
-            type.c_str());
-    return ML_ERROR_INVALID_PARAMETER;
-  }
+  // FIXME: why such restriction exists
```

Comment:
I think this was related to support Inplace

Reply:
This is adding restriction on the first layer. If the user wants it, let them place it. 
Support inPlace should not be affected with this. 
Support inPlace must adjust itself than restricting the model architectures.

Reply:
Removed the commented code.